### PR TITLE
Add parallel read_excel feature

### LINKED ISSUE #1298: read_excel function implementation

### PR DESCRIPTION
**Pull request by devin-petersohn**
* Resovles #1298

Implementation at a high level:

* Parse header on driver.
* Compute partition offsets based on chunks. Ensure that entire rows are
  read by detecting `</row>` closing tag.
* Send header and args to each partition.
* Create fake spreadsheet with header and subset of rows.
* Parse with openpyxl.
* Create dataframe from parsed sheet with pandas.

Currently not implemented:

* Multiple sheets
* `engine` specified as anything but `openpyxl`
* Python 3.6 - need `Zipfile` capabilities only available in 3.7+

Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1298 
- [x] tests added and passing

### CODE REVIEW

File: modin/engines/base/io/text/excel_reader.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                    if len(column_names) > (column_chunksize * (i + 1))
+                    else 0
+                    if len(column_names) < (column_chunksize * i)
+                    else len(column_names) - (column_chunksize * i)
+                    for i in range(num_splits)
+                ]
+            kwargs["num_splits"] = num_splits
+
+            while f.tell() < total_bytes:
+                print(f.tell())
```
Comment (aregm) [inline modin/engines/base/io/text/excel_reader.py]: Why there is a need to print this?

File: modin/engines/base/io/text/excel_reader.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+                # Make sure we are reading at least one row.
+                while row_count == 0:
+                    chunk += f.read(chunk_size)
+                    row_count += re.subn(row_close_tag, b"", chunk)[1]
+
+                last_index = chunk.rindex(row_close_tag)
+                f.seek(-(len(chunk) - last_index) + len(row_close_tag), 1)
+                args["end"] = f.tell()
+                if break_condition:
```
Comment (aregm) [inline modin/engines/base/io/text/excel_reader.py]: break_condition is used only here, for readability may be better to put it here?

File: modin/engines/base/io/text/excel_reader.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        from openpyxl.worksheet.worksheet import Worksheet
+        from openpyxl.worksheet._reader import WorksheetReader
+        from openpyxl.reader.excel import ExcelReader
+        from modin.backends.pandas.parsers import PandasExcelParser
+
+        sheet_name = kwargs.get("sheet_name", 0)
+        if sheet_name is None or isinstance(sheet_name, list):
+            warnings.warn(
+                "`read_excel` functionality is only implemented for a single sheet at a "
+                "time. Multiple sheet reading coming soon!"
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/engines/base/io/text/excel_reader.py:55]: 😄  This is somewhat trivial to implement, but I didn't want to complicate first implementation.

File: modin/engines/base/io/text/excel_reader.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            kwargs["_header"] = sheet_header
+            footer = b"</sheetData></worksheet>"
+            # Use openpyxml to parse the data
+            reader = WorksheetReader(
+                ws, BytesIO(sheet_header + footer), ex.shared_strings, False
+            )
+            # Attach cells to the worksheet
+            reader.bind_cells()
+            data = PandasExcelParser.get_sheet_data(
+                ws, kwargs.pop("convert_float", True)
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/engines/base/io/text/excel_reader.py:103]: In some cases we want to convert float back to integer, this is the default behavior that pandas implements.

File: modin/engines/base/io/text/excel_reader.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            total_bytes = cls.file_size(f)
+
+            from modin.pandas import DEFAULT_NPARTITIONS
+
+            num_partitions = DEFAULT_NPARTITIONS
+            # Read some bytes from the sheet so we can extract the XML header and first
+            # line. We need to make sure we get the first line of the data as well
+            # because that is where the column names are. The header information will
+            # be extracted and sent to all of the nodes.
+            sheet_block = f.read(4096)
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/engines/base/io/text/excel_reader.py:82]: Yes and yes

**Pull request merged**